src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -689,7 +696,7 @@ private bool IsCandidateSymbol(ISymbol memberSymbol)
             }
 
             private bool IsEntryPoint(IMethodSymbol methodSymbol)
-                => methodSymbol.Name == WellKnownMemberNames.EntryPointMethodName &&
+                => (methodSymbol.Name == WellKnownMemberNames.EntryPointMethodName || methodSymbol.Name == "$Main") &&
```

Comment:
@AlekseyTs Ideally we would use a helper for "unspeakable names", but I couldn't find one. Can you point us in the right direction?

Reply:
> Ideally we would use a helper for "unspeakable names", but I couldn't find one. Can you point us in the right direction?

```SyntaxFacts.IsValidIdentifier```?

Reply:
You also mentioned ```ISymbol.CanBeReferencedByName``` offline. However, it doesn't feel right to include all methods with unspeakable names into entry points.

---
In reply to: [439097752](https://github.com/dotnet/roslyn/pull/45090#discussion_r439097752) [](ancestors = 439097752)